packages/dd-trace/src/encode/0.4.js

```diff
@@ -66,7 +71,8 @@ class AgentEncoder {
   _encode (bytes, trace) {
     this._encodeArrayPrefix(bytes, trace)
 
-    for (const span of trace) {
+    const events = trace.map(formatSpan)
+    for (const span of events) {
```

Comment:
A `.map()` just before a for loop ?  I would have just emulated the map inside the for loop, so we don't iterate on it twice:
```suggestion
    for (let span of trace) {
      span = formatSpan(span)
```